pkg/docker/docker.go

```diff
@@ -310,14 +318,20 @@ func NewEngineAPIClient(config *api.DockerConfig) (*dockerapi.Client, error) {
 			},
 		}
 	}
-	return dockerapi.NewClient(config.Endpoint, os.Getenv("DOCKER_API_VERSION"), httpClient, nil)
+
+	return dockerapi.NewClientWithOpts(
+		dockerapi.WithHost(config.Endpoint),
+		dockerapi.WithHTTPClient(httpClient),
+		dockerapi.WithAPIVersionNegotiation(),
```

Comment:
My understanding is that `APIVersionNegotiation` is more or less incompatible with `FromEnv` if the `DOCKER_API_VERSION` env var is set.

See https://github.com/moby/moby/blob/24.0/client/client_deprecated.go#L13-L17

Reply:
I think `WithAPIVersionNegotiation()` is necessary. It does not work on my system with older moby.
Of course if `DOCKER_API_VERSION` is set that will take precedence.

Reply:
so `DOCKER_API_VERSION` beats negotiation, but from my experience we need negotiation.

Reply:
IMO this is expected and desired behaviour.